pkg/rke2/psp.go

```diff
@@ -324,16 +325,46 @@ func setPSPs(clx *cli.Context) func(context.Context, daemonsConfig.Control) erro
 				}
 			}
 
-			if _, err := cs.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{}); err != nil {
-				logrus.Fatalf("psp: update namespace annotation: %s", err.Error())
+			if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				if _, err := cs.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{}); err != nil {
+					if apierrors.IsConflict(err) {
+						if err := updateNamespace(ctx, cs, ns); err != nil {
+							return err
+						}
+					}
+					return err
+				}
+				return nil
+			}); err != nil {
+				logrus.Fatalf("psp: update namespace: %s - %s", ns.Name, err.Error())
 			}
-
 			logrus.Info("Applying PSP's complete")
 		}()
 		return nil
 	}
 }
 
+// updateNamespace receives a value of type v1.Namespace pointer
+// and updates that value to point to a newly retrieve value in
+// the event a conflict error is returned.
+func updateNamespace(ctx context.Context, cs *kubernetes.Clientset, ns *v1.Namespace) error {
```

Comment:
with this func name i was expecting to see an `cs.CoreV1().Namespaces().Update()` invocation. can we rename to `getNamespace` or `cacheNamespace`, etc?